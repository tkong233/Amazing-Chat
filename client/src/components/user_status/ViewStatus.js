import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import "./ViewStatus.css";
import Stories from "react-insta-stories";
import { Avatar } from "@material-ui/core";
import axios from "axios";

const ViewStatus = (props) => {
  const [status, setStatus] = useState("");
  const { email } = props.user;
  // const statusList = [];
  useEffect(() => {
    axios
      .get(`/status/${email}`)
      .then((res) => {
        setStatus(convertToDiv(res.data));
        // console.log(res.data);
      })
      .catch((err) => {
        console.log(err);
      });
  },[]);

  return (
    <div className="ViewStatus">
      <div className="stories">
        {status && 
          <Stories
            loop={false}
            width={600}
            keyboardNavigation
            defaultInterval={2000}
            stories={status}
            onStoryEnd={(s, st) => {
              console.log(st.id);
            }}
            // onStoryEnd={(s, st) => console.log("story ended", s, st)}
            onAllStoriesEnd={(s, st) => console.log("all stories ended", s, st)}
            onStoryStart={(s, st) => console.log("story started", s, st)}
          />
        }
      </div>
    </div>
  );
};

const convertToDiv = (data) => {
  console.log(data);
  let list = [];
  data.forEach((status) => {
    list.push({
      content: () => {
        if (status.statusData.image) {
          return (
            <div
              style={{
                ...contentStyle,
                backgroundImage: `url(${status.statusData.image})`,
              }}
            >
              <div style={{ display: "flex" }}>
                <Avatar style={{}} src={status.profile_picture} alt="" />
                <h6 style={{ marginLeft: "10px" }}>{status.name}</h6>
              </div>
  
              <div style={{ position: "absolute", bottom: 50 }}>
                <h4>
                {status.statusData.text}
                </h4>
              </div>
            </div>
          );
        } else {
          return (
            <div
              style={{ ...contentStyle }}
            >
              <div style={{ display: "flex" }}>
                <Avatar style={{}} src={status.profile_picture} alt="" />
                <h6 style={{ marginLeft: "10px" }}>{status.name}</h6>
              </div>
  
              <div style={{ position: "absolute", bottom: 50 }}>
                <h4>
                {status.statusData.text}
                </h4>
              </div>
            </div>
          );
        }
        
      },
      id: status.statusData.statusId,
    });
  });
  return list;
};

const contentStyle = {
  background: "#546e7a",
  width: "100%",
  padding: 20,
  color: "white",
  textShadow: "0 0 1px #000",
  position: "relative",
  backgroundSize: "cover",
  backgroundPosition: "center center",
};

const mapStateToProps = (state) => ({
  user: state.auth.user,
});

export default connect(mapStateToProps, {})(ViewStatus);
