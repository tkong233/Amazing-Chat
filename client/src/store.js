import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from 'redux-persist'
import rootReducer from "./reducers";
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

const initialState = {};

const persistConfig = {
  key: 'root',
  storage,
}

const middleware = [thunk];

const persistedReducer = persistReducer(persistConfig, rootReducer, initialState);

const store = createStore(

  persistedReducer,
  compose(
    applyMiddleware(...middleware),
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()) ||
      compose
  )
);

const persistor = persistStore(store);

export {
  store,
  persistor,
  middleware
};



