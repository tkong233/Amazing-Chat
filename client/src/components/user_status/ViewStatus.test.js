import React from 'react';
import ViewStatus from './ViewStatus';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 
import { applyMiddleware, createStore } from 'redux';
import {middleware} from '../../store';
import rootReducer from '../../reducers/index'

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});

const setUp = (initialState={}) =>{

    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    const store = createStoreWithMiddleware(rootReducer, initialState);
    const wrapper = shallow(<ViewStatus store={store} />).dive();
    // console.log(wrapper.debug());
    return wrapper;
}

describe('ViewStatus Component', ()=>{
    describe('Checking PropTypes', ()=>{
        it('Should not throw a warning', ()=>{
            const expectedProps = {
                user: {},
            }
            // eslint-disable-next-line react/forbid-foreign-prop-types
            const propsErr = checkPropTypes(ViewStatus.propTypes, expectedProps, 'props', ViewStatus.name);
            expect(propsErr).toBeUndefined();
        })
    });
    describe('Renders', ()=>{
        let wrapper;
        beforeEach(()=>{
            const initialState = {
                auth:{user: {
                    name: 'testuser',
                    email: 'testuser@test.com',
                }}
            };
            wrapper = setUp(initialState);
        });

        it('Should render without errors', ()=>{
            const container = wrapper.find(`[data-test='ViewStatusComponent']`);
            expect(container.length).toBe(1);
        });

        // it('Should status', ()=>{
        //     const container2 = wrapper.find(`[data-test='ViewStatusComponent2']`);
        //     expect(container2.length).toBe(1);
        // });

        it('Snapshot testing', ()=>{
            expect(wrapper).toMatchSnapshot();
        });

    });
});