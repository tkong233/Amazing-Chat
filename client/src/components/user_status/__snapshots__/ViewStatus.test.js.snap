// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewStatus Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
